### PR TITLE
Update type definition

### PR DESCRIPTION
**Pull request by jerry4718**

### CODE REVIEW

**Comment by januscla** [comment]
Thanks for your contribution, @jerry4718! Please make sure you sign our [CLA](https://cla.conf.meetecho.com/janus), as it's a required step before we can merge this.

**Comment by lminiero** [comment]
There's a lot of changes here. Can you please elaborate on what led you to come up with these changes, and explain them? Thanks!

**Comment by lminiero** [comment]
Pinging @atoppi as this is beyond my poor JS skills :hand_over_mouth:

File: npm/janus.d.ts [hunk 1]
```diff
@@ -241,7 +343,7 @@ declare namespace JanusJS {
 		getServer(): string;
 		isConnected(): boolean;
 		reconnect(callbacks: ReconnectOptions): void;
-		getSessionId(): number;
+		getSessionId(): string;
```
Comment (atoppi) [inline npm/janus.d.ts:346]: `number`

**Review by atoppi — COMMENTED** [review]
Thanks for the effort!
I've left some comments.

File: npm/janus.d.ts (hunk 1 above)
Comment (jerry4718) [reply npm/janus.d.ts:346]: I am really sorry that I have so many negligence, then I will address these points in detail next, thanks for the review

**Comment by jerry4718** [comment]
> Thanks for the effort! I've left some comments.

I am really sorry that I have so many negligence, then I will address these points in detail next, thanks for the review

File: npm/janus.d.ts [hunk 2]
```diff
@@ -1,30 +1,78 @@
...
+		error: (error: string, reason?: unknown) => void,
+	}
+
+	type JanusRequest = {
+		plugin?: string,
+		token?: string,
+		apisecret?: string,
+		session_id?: number,
+		handle_id?: number,
+		opaque_id?: number,
```
Comment (atoppi) [inline npm/janus.d.ts:57]: `opaque_id` is a string

File: npm/janus.d.ts [hunk 3]
```diff
@@ -1,29 +1,79 @@
...
+		plugin?: string,
+		token?: string,
+		apisecret?: string,
+		session_id?: string,
+		handle_id?: string,
+		opaque_id?: string,
+		loop_index?: string,
+		janus: string,
+		transaction: string,
+		body?: any | string
```
[the remark this answers is not in the document]
Comment (atoppi) [reply npm/janus.d.ts:61]: I still see this as possible string

File: npm/janus.d.ts [hunk 4]
```diff
@@ -187,12 +241,84 @@ declare namespace JanusJS {
...
+		error?: (error: string) => void;
+	}
+
+	type TrackOption = {
+		add: boolean;
+		replace: boolean;
+		remove: boolean;
+		type: 'video' | 'screen' | 'audio' | 'data';
+		mid: string;
+		capture: MediaStreamTrack;
```
Comment (atoppi) [inline npm/janus.d.ts:275]: this could also be `boolean`

File: npm/janus.d.ts [hunk 5]
```diff
@@ -187,12 +241,84 @@ declare namespace JanusJS {
...
+		recv: boolean;
+		group: 'default' | string;
+		gumGroup: TrackOption['group'];
+		simulcast: unknown;
```
Comment (atoppi) [inline npm/janus.d.ts:279]: `boolean`

File: npm/janus.d.ts [hunk 6]
```diff
@@ -187,12 +241,84 @@ declare namespace JanusJS {
...
+		svc: string;
+		simulcastMaxBitrates: unknown;
```
Comment (atoppi) [inline npm/janus.d.ts:281]: object with `number` values and keys
- low
- medium
- high

File: npm/janus.d.ts [hunk 7]
```diff
@@ -187,12 +241,84 @@ declare namespace JanusJS {
...
+		sendEncodings: unknown;
```
Comment (atoppi) [inline npm/janus.d.ts:282]: [RTCRtpEncodingParameters](https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpEncodingParameters)

File: npm/janus.d.ts [hunk 8]
```diff
@@ -187,12 +241,84 @@ declare namespace JanusJS {
...
+		framerate: number;
+		bitrate: number;
+		dontStop: boolean;
+		transforms: {
+			sender: ReadableWritablePair;
```
Comment (atoppi) [inline npm/janus.d.ts:287]: I guess this is a typescript type with `readable` and `writable` fields ?

**Review by atoppi — CHANGES_REQUESTED** [review]

**Review by atoppi — APPROVED** [review]
now it looks good !

**Comment by Pratikh** [comment]
Hi

We are using Janus in our project with TS so just wanted to know how can use type definition which you defined?


Janus is awesome Thanks @lminiero

**Pull request merged**